Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -438,6 +439,36 @@ def check_if_supported_sdk_available
           #{update_instructions}
         EOS
       end
+
+      # The CLT 10.x -> 11.x upgrade process on 10.14 contained a bug which broke the SDKs.
+      # Notably, MacOSX10.14.sdk would indirectly symlink to MacOSX10.15.sdk.
+      # This diagnostic was introduced to check for this and recommend a full reinstall.
+      def check_broken_sdks
+        locator = MacOS.sdk_locator
+
+        return if locator.all_sdks.all? do |sdk|
+          path_version = sdk.path.basename.to_s[/MacOSX(\d+\.\d+)\.sdk$/, 1]
+          next true if path_version.blank?
+
+          path_version = Version.new(path_version)
+          sdk.version == if path_version.major >= 11
+            path_version.major.to_s
+          else
+            path_version.to_s
+          end
+        end
+
+        source = if locator.source == :clt
+          "Command Line Tools (CLT)"
+        else
+          "Xcode"
+        end
+
+        <<~EOS
+          The SDKs in your #{source} installation are broken and require reinstallation.
```

Comment:
Can you be more specific about what "broken" means here and also how they should reinstall.

Reply:
Yeah, this message is rubbish. I'll rewrite it.